src/System.Net.Primitives/tests/PerformanceTests/IPAddressPerformanceTests.cs

```diff
@@ -0,0 +1,116 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.Net.Primitives.Tests
+{
+    public static class IPAddressPerformanceTests
+    {
+        public static readonly object[][] TestAddresses =
+        {
+            new object[] { new byte[] { 0x8f, 0x18, 0x14, 0x24 } },
+            new object[] { new byte[] { 0x10, 0x20, 0x30, 0x40, 0x50, 0x60, 0x70, 0x80, 0x90, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16 } },
+        };
+
+        [Benchmark]
+        [MeasureGCCounts]
+        [MemberData(nameof(TestAddresses))]
+        public static void TryWriteBytes(byte[] address)
+        {
+            var ip = new IPAddress(address);
+            var bytes = new byte[address.Length];
+            var bytesSpan = new Span<byte>(bytes);
+            int bytesWritten = 0;
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    ip.TryWriteBytes(bytesSpan, out bytesWritten);
```

Comment:
You have some test data in this PR, the numbers are very small so possibly too little for meaningsful results. Isn't it possible to increase iteration counts per method?